lib/connection.js

```diff
@@ -377,6 +397,8 @@ class Connection extends EventEmitter {
       this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this)
     );
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
@lundibundi if you put it this way, I agree with @belochub because I see no use in emiting credential in release mode.